moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/trajectory_visualization.hpp

```diff
@@ -137,7 +137,7 @@ private Q_SLOTS:
   /**
    * \brief ROS callback for an incoming path message
    */
-  void incomingDisplayTrajectory(const moveit_msgs::msg::DisplayTrajectory::ConstPtr& msg);
+  void incomingDisplayTrajectory(const moveit_msgs::msg::DisplayTrajectory::SharedPtr msg);
```

Comment:
Look at this https://answers.ros.org/question/212857/what-is-constptr/

Reply:
ConstPr was throwing warnings regarding a current or future deprecated state, not sure. Maybe we should replace `ConstPtr` for `const X SharedPtr` ?

Reply:
Williams says on the link that I sent you that a replacement for a Ptr shoud be as following:

``
std_msgs::msg::String::Ptr
`` to `` std::shared_ptr<std_msgs::msg::String>`` or ``std_msgs::msg::String::SharedPtr``

and with constPtr
`` std::shared_ptr<const std_msgs::msg::String>`` It says that using the const before the msg sometimes is needed because for *std_msgs::string* it might not make much difference, but for example for a *sensor_msgs::PointCloud2* it might.